src/Util/ParametersManipulator.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Util;
+
+/**
+ * @author Willem Verspyck
+ */
+final class ParametersManipulator
+{
+    /**
+     * Merge parameters, but replace them when it's a subarray.
+     */
+    public static function merge(array $parameters, array $filters): array
```

Comment:
Given the class name and the method name, it seems too agnostic for an argument called `$filter`.
I mean, here is no visible association with the "filter" concept.

Reply:
I understand. Will $parametersToMerge be better than $filters? I will try to update this PR tomorrow.

Reply:
WDYT of
```
merge(array $parameters, array $newParameters)
```